src/components/views/DownloadModModal.vue

```diff
@@ -184,47 +184,35 @@ import ProfileModList from '../../r2mm/mods/ProfileModList';
             }, 1);
         }
 
-        static async installModAfterDownload(profile: Profile, mod: ThunderstoreMod, version: ThunderstoreVersion): Promise<R2Error | void> {
-            return new Promise(async (resolve, reject) => {
+        static async installModAfterDownload(profile: Profile, mod: ThunderstoreMod, version: ThunderstoreVersion): Promise<void> {
+            const profileModList = await ProfileModList.getModList(profile.asImmutableProfile());
+            if (profileModList instanceof R2Error) {
```

Comment:
Could use `throwForR2Error` here?

Reply:
Didn't do it at first, because I couldn't figure out how to make typescript happy with that, but now I did. :D

Reply:
Ah, right. If you would've mentioned it messes with type inference I would've probably told you to not bother with it 😅